packages/db/index.js

```diff
@@ -935,7 +936,9 @@ export class DBClient {
       query = query.gte('inserted_at', opts.after)
     }
 
-    // TODO(https://github.com/web3-storage/web3.storage/issues/798): filter by meta is missing
+    if (opts.meta) {
+      query = query.eq('meta', opts.meta)
```

Comment:
I thinks this will not be enough per https://ipfs.github.io/pinning-services-api-spec/#section/Custom-metadata/Pin-metadata

- lookup returns pins that have meta with all key-value pairs matching the passed values
- pin metadata may have more keys, but only ones passed in the query are used for filtering

Reply:
Now corrected, filters on all the key/values passed.